Update versions

davidbuzinski opened the pull request.

davidbuzinski committed:
update versions

davidbuzinski committed:
update actions version

davidbuzinski committed:
Update azure-pipelines.yml

mw-hrastega commented:
> _ No description provided. _

Should README.md also be part of this pull request?

davidbuzinski commented:
> > _ No description provided. _
> 
> Should README.md also be part of this pull request?

Yes, it's included. I just updated the versions in the examples.

davidbuzinski committed:
Update README.md

mw-hrastega commented on README.md:
```diff
@@ -26,25 +26,39 @@ Each of these pipeline definitions does four things:
       sourceFolder: source
       codeCoverageCobertura: coverage.xml
-  - script: bash <(curl -s https://codecov.io/bash)
+  - script: |
```
The original code was terse. This replacement is quite verbose and hard to understand. Can I know why we had to make this replacement?

davidbuzinski replied:
It looks like the bash script is still available, but they are moving towards using their CLI going forward. When you setup a new project on codecov.io and choose "Other CI" (Not GitHub Actions or CircleCI) they recommend doing things this way
<img width="786" alt="Screenshot 2024-04-02 at 11 10 25 AM" src="https://github.com/mathworks/matlab-codecov-example/assets/103441853/1f1c7fc1-4f37-4aa0-9542-dc60b463485f">
mend this code:

mw-hrastega replied:
Good to know. Thanks for clarifying, David.

mcafaro commented on README.md:
```diff
@@ -26,25 +26,39 @@ Each of these pipeline definitions does four things:
       sourceFolder: source
       codeCoverageCobertura: coverage.xml
-  - script: bash <(curl -s https://codecov.io/bash)
+  - script: |
+      # download Codecov CLI
+      curl -Os https://cli.codecov.io/latest/linux/codecov
+      
+      # integrity check
+      curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step  
+      curl -Os https://cli.codecov.io/latest/linux/codecov
+      curl -Os https://cli.codecov.io/latest/linux/codecov.SHA256SUM
+      curl -Os https://cli.codecov.io/latest/linux/codecov.SHA256SUM.sig
+      gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+
+      # upload to Codecov 
+      shasum -a 256 -c codecov.SHA256SUM
+      sudo chmod +x codecov
+      ./codecov upload-process -t $(CODECOV_TOKEN)
```
Why not use the codecov action? https://github.com/marketplace/actions/codecov

mw-hrastega commented:
> > > _ No description provided. _
> > 
> > 
> > Should README.md also be part of this pull request?
> 
> Yes, it's included. I just updated the versions in the examples.

Ok, I think some text also needs to be updated:

1) You wanted to discontinue support for Travis CI. Does it still make sense to have it on the page?
2) The Caveats section is outdated.

davidbuzinski committed:
remove travis and caveats

davidbuzinski commented:
> > > > _ No description provided. _
> > > 
> > > 
> > > Should README.md also be part of this pull request?
> > 
> > 
> > Yes, it's included. I just updated the versions in the examples.
> 
> Ok, I think some text also needs to be updated:
> 
>     1. You wanted to discontinue support for Travis CI. Does it still make sense to have it on the page?
> 
>     2. The Caveats section is outdated.

Good idea. I removed the Travis example and caveats section.

davidbuzinski committed:
Update README.md

davidbuzinski committed:
Update README.md
